Sources/NIOSSL/SSLContext.swift

```diff
@@ -112,6 +112,101 @@ private func alpnCallback(ssl: OpaquePointer?,
     return SSL_TLSEXT_ERR_OK
 }
 
+/// PSK Callback for the server side context.
+private func serverPSKCallback(ssl: OpaquePointer?,
+                                 identity: UnsafePointer<CChar>?,
+                                 psk: UnsafeMutablePointer<UInt8>?,
+                                 max_psk_len: UInt32) -> UInt32 {
+    
+    guard let ssl = ssl else { return 0 }
+    
+    // Initial implementation only supports TLS 1.2 due API support exposed from BoringSSL.
+    // TODO (meaton) add TLS 1.3 support when available.
+    
+    // We want to take the SSL pointer and extract the parent Swift object.
+    let parentCtx = CNIOBoringSSL_SSL_get_SSL_CTX(ssl)!
+    let parentPtr = CNIOBoringSSLShims_SSL_CTX_get_app_data(parentCtx)!
+    let parentSwiftContext: NIOSSLContext = Unmanaged.fromOpaque(parentPtr).takeUnretainedValue()
+    
+    guard let serverCallback = parentSwiftContext.pskServerConfigurationCallback,
+          let unwrappedIdentity = identity, // Incoming identity
+          let strIdentity = String(validatingUTF8: unwrappedIdentity),
+          let configurationHint = parentSwiftContext.configuration.pskHint,
+          let outputPSK = psk // Output PSK key.
+          else {
+        return 0
+    }
+    
+    // Take the hint and the possible identity and pass them down to the callback to get associated PSK from callback
+    guard let contextPSKIdentityCallback = try? serverCallback(configurationHint, strIdentity) else {
+        return 0
+    }
+    let serverPSK = contextPSKIdentityCallback.key // From the callback
+    
+    // Make sure the key is returned by the callback and it is of proper length, otherwise, fail.
+    if serverPSK.isEmpty || serverPSK.count > max_psk_len {
+        return 0
+    }
+    let _ = serverPSK.withUnsafeBytes { (body: UnsafeRawBufferPointer) -> Void in
+        guard let unsafeRawPointer = body.baseAddress else { return }
+        memcpy(outputPSK, unsafeRawPointer, body.count)
+    }
+    return UInt32(serverPSK.count)
```

Comment:
This also needs to use `serverPSK.utf8.count`.

Reply:
In this context `serverPSK` is `NIOSSLSecureBytes` and not a string, so count would work here, no?